merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/timeline/TemporalEventSource.java

```diff
@@ -86,4 +86,8 @@ public sealed interface TimePoint {
     record Delta(Duration delta) implements TimePoint {}
     record Commit(EventGraph<Event> events, Set<Topic<?>> topics) implements TimePoint {}
   }
+
+  public void freeze() {
+    this.points.freeze();
+  }
```

Comment:
Tentative: because the `freeze` is delegated to `this.points`, a `SlabList` that can be frozen. As soon as it will try to append something to `this.points`, an exception will be thrown.